gitlab/base.py

```diff
@@ -323,6 +324,21 @@ class RequiredOptional:
     optional: Tuple[str, ...] = ()
 
 
+def custom_attrs(required: tuple = (), optional: tuple = ()) -> Callable[[F], F]:
+    """Decorates a custom method to add a RequiredOptional attribute.
+
+    Args:
+        required: A tuple of API attributes required in the custom method
+        optional: A tuple of API attributes optional in the custom method
+    """
+
+    def decorator(func: F) -> F:
+        setattr(func, "_custom_attrs", RequiredOptional(required, optional))
```

Comment:
Can it be called `_required_optional` instead of `_custom_attrs`?

Reply:
Hehe I actually started with exactly that :) But I changed my mind later, see above:

> So I wanted to make them more similar to `_create_attrs` and `_update_attrs` class attributes, which currently have the exact same structure. WDYT?

Hope that makes sense.